src/types/productTypes.ts

```diff
@@ -0,0 +1,56 @@
+import { IState } from '@types';
+
+export interface IStock {
+  amount: number;
+  measure: BuyBy;
+}
+
+export interface IPrice {
+  amount: number;
+  discountedAmount: number;
+  discount: number;
+  currency: string;
+}
+
+export interface IDescription {
+  short: string;
+  long: string;
+  full: string;
+}
+
+export interface IDelivery {
+  timeframe: number;
+  cost: number;
+  area: string[];
+}
+
+export enum BuyBy {
+  PSC = 'pcs',
+  KGS = 'kgs',
+  BOX = 'box',
+  PACK = 'pack',
+}
+
```

Comment:
Think it's better to have file for each type in this case

Reply:
Decomposed the types to separate files.